omod/src/main/java/org/openmrs/module/datafilter/extension/html/Content.java

```diff
@@ -49,15 +51,18 @@ public String generate() {
 	private String addHTMLForLocationNames() {
 		return locationNames.stream().reduce("", (appendedContent, locationName) -> {
 			String checkedValue = "";
+			String escapedLocationName = StringEscapeUtils.escapeHtml(locationName);
```

Comment:
I don't think you need to introduce a new variable here, you can just do:
```java
locationName = StringEscapeUtils.escapeHtml(locationName);
```

Reply:
Hi @mks-d,
That would not work, because I still need the original value to validate if the location is selected.
Unless I move that validation up. If you prefer I can do that.

Reply:
@icrc-loliveira you mean this bit:
```java
if (selectedLocations.contains(locationName)) {
  checkedValue = CHECKED;
}
```
?
If yes, then yes move it up.

Reply:
Done.